subsys/bluetooth/controller/ll_sw/coex_ticker.c

```diff
@@ -0,0 +1,258 @@
+/*
+ * Copyright (c) 2020 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <errno.h>
+
+#include <zephyr.h>
+#include <device.h>
+#include <init.h>
+#include <soc.h>
+
+#include <logging/log.h>
+#include <drivers/gpio.h>
+#include <sys/__assert.h>
+#include <bluetooth/hci.h>
+#include "controller/hal/ticker.h"
+#include "controller/ticker/ticker.h"
+#include "controller/include/ll.h"
+
+LOG_MODULE_REGISTER(coex_ticker);
+
+#define COEX_RADIO_WORK_DELAY_US  50
+#define COEX_RADIO_WORK_RESCHEDULE_DELAY_US  200
+
+struct coex_ticker_config {
+	struct gpio_dt_spec inhibit_spec;
+	size_t inhibit_delay_us;
+};
+struct coex_ticker_data {
+	const struct device *dev;
+	struct gpio_callback inhibit_irq_cb;
+};
+
+static inline int _ticker_stop(uint8_t inst_idx, uint8_t u_id, uint8_t tic_id)
+{
+	int ret = ticker_stop(inst_idx, u_id, tic_id, NULL, NULL);
+
+	if (ret != TICKER_STATUS_SUCCESS &&
+		ret != TICKER_STATUS_BUSY) {
+		__ASSERT(0, "Failed to stop ticker.\n");
+	}
+
+	return ret;
+}
+
+static int time_slot_delay(uint32_t ticks_at_expire, uint32_t ticks_delay,
+				ticker_timeout_func callback, void *context)
+{
+	uint8_t instance_index;
+	uint8_t ticker_id;
+	int err;
+
+	ll_timeslice_ticker_id_get(&instance_index, &ticker_id);
+
+	/* start a secondary one-shot ticker after ticks_delay,
+	 * this will let any radio role to gracefully abort and release the
+	 * Radio h/w.
+	 */
+	err = ticker_start(instance_index, /* Radio instance ticker */
+				1, /* user id for link layer ULL_HIGH */
+				   /* (MAYFLY_CALL_ID_WORKER) */
+				(ticker_id + 1), /* ticker_id */
+				ticks_at_expire, /* current tick */
+				ticks_delay, /* one-shot delayed timeout */
+				0, /* periodic timeout  */
+				0, /* periodic remainder */
+				0, /* lazy, voluntary skips */
+				0,
+				callback, /* handler for executing radio abort or */
+					 /* flash work */
+				context, /* the context for the flash operation */
+				NULL, /* no op callback */
+				NULL);
+
+	return err;
+}
+
+
+#if defined(CONFIG_BT_CTLR_DEBUG_PINS) || \
+	defined(CONFIG_BT_CTLR_DEBUG_PINS_CPUAPP)
+#define DEBUG_PORT NRF_P0
+#define DEBUG_PIN_INHIBIT BIT(23)
+#define DEBUG_PIN_IRQ BIT(25)
+#define DEBUG_PIN_MASK    (DEBUG_PIN_IRQ | DEBUG_PIN_INHIBIT)
+#define DEBUG_INIT() \
+	do { \
+		DEBUG_PORT->DIRSET = DEBUG_PIN_MASK; \
+		DEBUG_PORT->OUTCLR = DEBUG_PIN_MASK; \
+	} while (0)
+
+#define DEBUG_INHIBIT(flag) \
+	do { \
+		if (flag) { \
+			DEBUG_PORT->OUTSET = DEBUG_PIN_INHIBIT; \
+		} else { \
+			DEBUG_PORT->OUTCLR = DEBUG_PIN_INHIBIT; \
+		} \
+	} while (0)
+
+
+#define DEBUG_IRQ(flag) \
+	do { \
+		if (flag) { \
+			DEBUG_PORT->OUTSET = DEBUG_PIN_IRQ; \
+		} else { \
+			DEBUG_PORT->OUTCLR = DEBUG_PIN_IRQ; \
+		} \
+	} while (0)
+#else
+#define DEBUG_INIT()
+#define DEBUG_INHIBIT(flag)
+#define DEBUG_IRQ(flag)
+#endif
+
```

Comment:
Add the debug macros here (`DEBUG_COEX_INHIBIT` and `DEBUG_COEX_IRQ`): https://github.com/zephyrproject-rtos/zephyr/blob/main/subsys/bluetooth/controller/ll_sw/nordic/hal/nrf5/debug.h#L321

```suggestion
```